endpoint.go

```diff
@@ -289,10 +290,14 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if err = sb.populateNetworkResources(ep); err != nil {
-		return err
+	sb.Lock()
```

Comment:
Also, I thought having the explicit check in Join makes it more easily readable for folks to know that they can still do the join without the actual need to have osSbox available.